From 44178a47297c1c72814486906933a2b29df3d5dc Mon Sep 17 00:00:00 2001
From: Manoj Srivastava <srivasta@golden-gryphon.com>
Date: Sat, 12 Apr 2014 16:07:43 -0700
Subject: [PATCH 1/1] [handle-missing-name]: Return undef unless a color name
 is provided

Perl 5.12.0 adds new uninitialized value warnings for the lc* and uc*
functions. One of these gets triggered by Graphics::ColorObject, making
libcolor-calc-perl fail its test suite with perl_5.12.0-1 on amd64.

There is indeed a fix in this commit (unlike the last attempt), by
returning early if no color name has been provided.

Signed-off-by: Manoj Srivastava <srivasta@golden-gryphon.com>
---
 ColorObject.pm | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/ColorObject.pm b/ColorObject.pm
index 1169f63..4a75558 100644
--- a/ColorObject.pm
+++ b/ColorObject.pm
@@ -1902,7 +1902,9 @@ our %COLORNAMES;
 tie %COLORNAMES, 'Graphics::ColorNames', qw(HTML Windows Netscape X);
 
 sub namecolor {
-	my $name=lc(shift @_);
+	my $name=shift;
+	return unless $name;
+	$name=lc $name;
 	$name=~s/[^\#!%\&a-z0-9]//g;
 	my $col;
 	my $opt=shift @_;
-- 
1.9.2

